### PR TITLE
Update scheduler and optimizer

### PR DESCRIPTION
**Pull request by alicjak1519**
According to the suggestion from the meeting, there are a couple of changes in the optimizers configuration.

### CODE REVIEW

**Comment by alicjak1519** [comment]
Screenshot 2022-03-15 at 10 23 20
According to Karol's suggestion, the scheduler and optimizer were updated. The green configuration, which seems to be the best, is optimizer: Adam, scheduler: OneCycleLR.

File: models/model.py [hunk 1]
```diff
@@ -183,33 +185,24 @@ def validation_epoch_end(self, out):
...
     def configure_optimizers(self):
-        # set optimizer
         optimizer = torch.optim.Adam(self.parameters(), lr=self.lr)
 
-        # set scheduler: multiply lr every epoch
-        def lambd(epoch):
-            return self.multiply_lr_step
-
-        self.scheduler = MultiplicativeLR(optimizer, lr_lambda=lambd)
+        steps_per_epoch = self.steps_per_epoch // self.trainer.accumulate_grad_batches
```
Comment (alicjak1519) [inline models/model.py:190]: According to the [comment](https://github.com/PyTorchLightning/pytorch-lightning/issues/1120#issuecomment-633641024)

**Review by mariaogryczak — APPROVED** [review]

**Review by natcz — APPROVED** [review]
Works for me (with reading the OneCycleLR docs knowledge).

**Review by AgaMiko — CHANGES_REQUESTED** [review]
I think that @karolmajek mentioned that onecyclelr has to be updated every step. Hence it should be moved from validation_epoch_end to validation_step. Can you check it?

**Comment by AgaMiko** [comment]
Oh sorry, it should be in the training step of course! My fault!

**Review by AgaMiko — APPROVED** [review]

**Comment by alicjak1519** [comment]
After doing some improvements suggested by @AgaMiko and after merging main into this branch, I conducted three experimental pieces of training to compare scheduler/optimizer options again. Due to the time limit, my plots represent only 1 epoch (so the advantages of the new scheduler and optimizer can not be very reliable). However, taking into account @karolmajek intuition, I would like to merge this PR. :)

Screenshot 2022-03-22 at 08 16 53
Screenshot 2022-03-22 at 08 16 58
